storage/blockdevice/source/SFDP.cpp

```diff
@@ -243,6 +243,7 @@ int sfdp_parse_sector_map_table(Callback<int(bd_addr_t, void *, bd_size_t)> sfdp
         tr_debug("No Sector Map Table");
 
         // If there's no sector map, we have a single region sized the entire device size
+        sfdp_info.smptbl.region_cnt = 1;
```

Comment:
Could you explain in the commit message for "SFDP: Set region count to 1 for no sector map" why this is needed and what problems we ran into before without this?

Reply:
I added this for consistency with the next two lines: `region_size[0]` and `region_high_boundary[0]` mean there's one region which is region 0. But `region_cnt` is previously initialised to 1 anyway, so I'm going to remove this commit.

Reply:
Ah, removing this commit results in failed unit test which (reasonably) expects `region_cnt` to be set to 1 by class SFDP, if there's only one region. I'll bring it back and improve commit message.